core/Test/Tasty/Ingredients/ConsoleReporter.hs

```diff
@@ -194,46 +195,27 @@ consoleOutput :: (?colors :: Bool) => TestOutput -> StatusMap -> IO ()
 consoleOutput toutput smap =
   getTraversal . fst $ foldTestOutput foldTest foldHeading toutput smap
   where
-    foldTest _name printName getResult printResult =
+    foldTest _name printName getResult printResult opts =
       ( Traversal $ do
-          printName :: IO ()
+          let HideSuccesses hideSuccesses = lookupOption opts
           r <- getResult
-          printResult r
+          _ <- if resultSuccessful r && hideSuccesses
+               then do pure $ Any False
+               else do printName :: IO (); printResult r :: IO (); return $ Any True
+          pure ()
       , Any True)
     foldHeading _name printHeading (printBody, Any nonempty) =
       ( Traversal $ do
           when nonempty $ do printHeading :: IO (); getTraversal printBody
       , Any nonempty
       )
 
-consoleOutputHidingSuccesses :: (?colors :: Bool) => TestOutput -> StatusMap -> IO ()
-consoleOutputHidingSuccesses toutput smap =
-  void . getApp $ foldTestOutput foldTest foldHeading toutput smap
-  where
-    foldTest _name printName getResult printResult =
-      Ap $ do
-          printName :: IO ()
-          r <- getResult
-          if resultSuccessful r
-            then do clearThisLine; return $ Any False
-            else do printResult r :: IO (); return $ Any True
-
-    foldHeading _name printHeading printBody =
-      Ap $ do
-        printHeading :: IO ()
-        Any failed <- getApp printBody
-        unless failed clearAboveLine
-        return $ Any failed
-
-    clearAboveLine = do cursorUpLine 1; clearThisLine
-    clearThisLine = do clearLine; setCursorColumn 0
-
 streamOutputHidingSuccesses :: (?colors :: Bool) => TestOutput -> StatusMap -> IO ()
 streamOutputHidingSuccesses toutput smap =
   void . flip evalStateT [] . getApp $
     foldTestOutput foldTest foldHeading toutput smap
   where
-    foldTest _name printName getResult printResult =
+    foldTest _name printName getResult printResult _opts =
```

Comment:
I think you should update this function too; otherwise your new feature will work only in terminals.